tokio-trace-tower-http/src/lib.rs

```diff
@@ -97,27 +125,45 @@ where
     type Error = T::Error;
 
     fn poll_ready(&mut self) -> futures::Poll<(), Self::Error> {
-        let span = &mut self.span;
-        let inner = &mut self.inner;
-        span.enter(move || inner.poll_ready())
+        if let Some(span) = &mut self.span {
+            let inner = &mut self.inner;
+            span.enter(move || inner.poll_ready())
+        } else {
+            self.inner.poll_ready()
+        }
     }
 
     fn call(&mut self, request: http::Request<B>) -> Self::Future {
-        let span2 = self.span.clone();
-        let span = &mut self.span;
-        let inner = &mut self.inner;
-        span.enter(move || {
-            span!(
-                Level::TRACE,
+        if let Some(span) = &mut self.span {
+            let span2 = span.clone();
+            let inner = &mut self.inner;
+            span.enter(move || {
+                trace_span!(
+                    "request",
+                    // TODO: custom `Value` impls for `http` types would be nicer
+                    // than just sticking these in `debug`s...
+                    method = &field::debug(request.method()),
+                    version = &field::debug(request.version()),
+                    uri = &field::debug(request.uri()),
+                    headers = &field::debug(request.headers())
+                )
+                .enter(move || inner.call(request).instrument(span2))
+            })
+        } else {
```

Comment:
I think this `if`/`else` logic could be much simpler if we used the `parent:` key in the `span!` macro instead. Consider the following:

```rust
fn call(&mut self, request: http::Request<B>) -> Self::Future {
    let inner = &mut self.inner;
    let span = trace_span!(
        parent: self.span.as_ref(),
        "request",
        // TODO: custom `Value` impls for `http` types would be nicer
        // than just sticking these in `debug`s...
        method = &field::debug(request.method()),
        version = &field::debug(request.version()),
        uri = &field::debug(request.uri()),
        headers = &field::debug(request.headers())
    ).enter(move || inner.call(request).instrument(span))
}
```